src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsClientModelValidator.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// A default implementation of <see cref="IClientModelValidator"/>.
+    /// </summary>
+    public class DataAnnotationsClientModelValidator : IClientModelValidator
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="DataAnnotationsClientModelValidator"/>.
+        /// </summary>
+        /// <param name="attribute">The <see cref="ValidationAttribute"/> which is used to get
+        /// <see cref="ModelClientValidationRule"/>s.</param>
+        public DataAnnotationsClientModelValidator([NotNull] ValidationAttribute attribute)
+        {
+            Attribute = attribute;
+        }
+
+        /// <summary>
+        /// The <see cref="ValidationAttribute"/> associated with this instance.
+        /// </summary>
+        public ValidationAttribute Attribute { get; private set; }
+
+        /// <inheritdoc />
+        public virtual IEnumerable<ModelClientValidationRule> GetClientValidationRules(
+            [NotNull] ClientModelValidationContext context)
+        {
+            var customValidator = Attribute as IClientModelValidator;
```

Comment:
> Is this needed because of DataTypeAttributeAdapter
No, `DataTypeAttributeAdapter` is already an `IClientModelValidator'.

You would need the cast as the type `ValidationAttribute`.
Moreover it good to have this as there as someone can create a custom attribute which does not implement `IClientModelValidator`.